Detectors/EMCAL/workflow/src/RawToCellConverterSpec.cxx

```diff
@@ -0,0 +1,149 @@
+// Copyright CERN and copyright holders of ALICE O2. This software is
+// distributed under the terms of the GNU General Public License v3 (GPL
+// Version 3), copied verbatim in the file "COPYING".
+//
+// See http://alice-o2.web.cern.ch/license for full licensing information.
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+#include "FairLogger.h"
+
+#include "DataFormatsEMCAL/EMCALBlockHeader.h"
+#include "DataFormatsEMCAL/TriggerRecord.h"
+#include "EMCALWorkflow/RawToCellConverterSpec.h"
+#include "Framework/ControlService.h"
+#include "SimulationDataFormat/MCCompLabel.h"
+#include "SimulationDataFormat/MCTruthContainer.h"
+#include "EMCALBase/Geometry.h"
+#include "EMCALBase/Mapper.h"
+#include "EMCALReconstruction/CaloFitResults.h"
+#include "EMCALReconstruction/Bunch.h"
+#include "EMCALReconstruction/CaloRawFitterStandard.h"
+#include "EMCALReconstruction/AltroDecoder.h"
+#include "CommonDataFormat/InteractionRecord.h"
+
+using namespace o2::emcal::reco_workflow;
+
+void RawToCellConverterSpec::init(framework::InitContext& ctx)
+{
+  LOG(DEBUG) << "[EMCALRawToCellConverter - init] Initialize converter ";
+  if (!mGeometry) {
+    mGeometry = o2::emcal::Geometry::GetInstanceFromRunNumber(223409);
+  }
+  if (!mGeometry)
+    LOG(ERROR) << "Failure accessing geometry";
+
+  if (!mMapper) {
+    mMapper = std::unique_ptr<o2::emcal::MappingHandler>(new o2::emcal::MappingHandler);
+  }
+  if (!mMapper)
+    LOG(ERROR) << "Failed to initialize mapper";
+
+  mRawFitter.setAmpCut(mNoiseThreshold);
+  mRawFitter.setL1Phase(0.);
+}
+
+void RawToCellConverterSpec::run(framework::ProcessingContext& ctx)
+{
+  LOG(DEBUG) << "[EMCALRawToCellConverter - run] called";
+  auto dataref = ctx.inputs().get("raw");
+  auto const* emcheader = o2::framework::DataRefUtils::getHeader<o2::emcal::EMCALBlockHeader*>(dataref);
+  if (!emcheader->mHasPayload) {
+    LOG(DEBUG) << "[EMCALCellConverter - run] No more raw data" << std::endl;
+    ctx.services().get<o2::framework::ControlService>().readyToQuit(framework::QuitRequest::Me);
+    return;
+  }
```

Comment:
This won't work: The raw to cell converter is subscribing directly to readout. There is no EMCAL block header but only the plain raw data. The whole part needs to be removed.

The EMCAL block header is rather intended to notify other components that there is payload data coming. This was based on the first version of the TPC reconstruction workflow. In such case the component would be responsible to send an EMCAL block header. However there were changes in the handling of the workflow termination in #3534 and #3542 which now uses framework level handling of the end-of-stream, which makes the block header superfluous. The DPL components needs to be adapted.